### PR TITLE
Cleaned Cosserat in tensor_mechanics

### LINKED ISSUE #6971: Improve documentation of Cosserat
### Description of the enhancement or error report

The Cosserat-related things in TensorMechanics have little (or zero) doxygen documentation.

### Rationale for the enhancement or information for reproducing the error

Jolly hard to understand what's going on at present!

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

No adverse impact

### PR DESCRIPTION
**Pull request by WilkAndy**
Included more doxygen comments
Made displacements and Cosserat_rotations vector inputs
Various minor tidying

Fixes #6971

### CODE REVIEW

**Comment by WilkAndy** [comment]
Hey i like these extra tests (valgrind).  i wonder why the "heavy" modules tests aren't automatic.

File: modules/tensor_mechanics/include/kernels/MomentBalancing.h [hunk 1]
```diff
@@ -27,18 +37,29 @@ class MomentBalancing : public Kernel
...
-  const unsigned int _ydisp_var;
-  const unsigned int _zdisp_var;
+  // number of Cosserat rotation components provided by the user
+  const unsigned int _nrots;
+
+  /// the moose variable numbers for the Cosserat rotation degrees of freedom
+  std::vector<unsigned int> _wc_var;
+
+  // number of displacement components provided by the user
+  const unsigned int _ndisp;
```
Comment (dschwen) [inline modules/tensor_mechanics/include/kernels/MomentBalancing.h]: Can ndisp be different than the number of rotation variables?

File: modules/tensor_mechanics/src/kernels/CosseratStressDivergenceTensors.C [hunk 2]
```diff
@@ -16,35 +16,25 @@ template<>
 InputParameters validParams<CosseratStressDivergenceTensors>()
 {
   InputParameters params = validParams<StressDivergenceTensors>();
-  params.addCoupledVar("wc_x", "The Cosserat rotation about x");
-  params.addCoupledVar("wc_y", "The Cosserat rotation about y");
-  params.addCoupledVar("wc_z", "The Cosserat rotation about z");
+  params.addRequiredCoupledVar("Cosserat_rotations", "The 3 Cosserat rotation variables");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/CosseratStressDivergenceTensors.C]: Doco asks for 3, but fewer are OK, too. Aren't those numbers always dim?

**Comment by permcody** [comment]
The valgrind tests aren't automatic simply due to resources. Most of those Valgrind tests take over an hour to run. The heaviest ones around three hours. We've put the switch there so that we can run them for certain PRs.

I'm not opposed to developers activating them as needed. Your time zone difference makes this a reasonable alternative. I don't think any of us would notice that you are running tons of Valgrind tests late in the evening here.

**Comment by sapitts** [comment]
Thank you for cleaning up this kernel @WilkAndy

File: modules/tensor_mechanics/include/kernels/MomentBalancing.h (hunk 1 above)
Comment (WilkAndy) [inline modules/tensor_mechanics/include/kernels/MomentBalancing.h]: At the moment, ndisp and nrots must both equal 3.  However, if the crowds of users clamor for a 2D version then ndisp=2 and nrots=1.

Comment (dschwen) [inline modules/tensor_mechanics/include/kernels/MomentBalancing.h]: Ah, ok, that makes sense. Thanks.

File: modules/tensor_mechanics/tests/static_deformations/tests [hunk 3]
```diff
@@ -11,4 +11,12 @@
...
   [../]
+
+  [./except01]
+    type = RunException
+    input = cosserat_tension.i
+    cli_args = 'Kernels/x_moment/component=1 Outputs/file_base=except01'
+    expect_err = 'MomentBalancing: The variable for this Kernel must be equal to the Cosserat rotation variable defined by the "component" and the "Cosserat_rotations" parameters'
+    prereq = tension
+  [../]
 []
```
Comment (permcody) [inline modules/tensor_mechanics/tests/static_deformations/tests]: Super nit-picky but would you mind adding a blank line here? We need to start scanning these "tests" files for this problem like we do everything else.

Comment (WilkAndy) [inline modules/tensor_mechanics/tests/static_deformations/tests]: Sure, no problems.

**Comment by permcody** [comment]
👍

**Comment by WilkAndy** [comment]
Thanks @permcody,

a

**Pull request merged**